src/main/java/edu/harvard/iq/dataverse/DvObjectContainer.java

```diff
@@ -56,4 +67,32 @@ public void setStorageDriverId(String storageDriver) {
             this.storageDriver = storageDriver;
         }
     }
+    
+    public String getEffectiveMetadataLanguage() {
+        String ml = metadataLanguage;
+        if (StringUtils.isBlank(ml)) {
+            if (this.getOwner() != null) {
+                ml = this.getOwner().getEffectiveMetadataLanguage();
+            } else {
+                ml = DEFAULT_METADATA_LANGUAGE_CODE;
+            }
+        }
+        return ml;
+    }
+    
+    public String getMetadataLanguage() {
+        if (metadataLanguage == null) {
+            return UNDEFINED_METADATA_LANGUAGE_CODE;
+        }
+        return metadataLanguage;
+    }
+
+    public void setMetadataLanguage(String ml) {
+        if (ml != null && ml.equals(UNDEFINED_METADATA_LANGUAGE_CODE)) {
```

Comment:
Is there a reason to not just allow it to be null for undefined instead of transforming back and forth in the getter and setter?